src/test/java/org/logstash/input/DeadLetterQueueInputPluginTests.java

```diff
@@ -112,4 +112,27 @@ public void testTimestamp() throws Exception {
         DeadLetterQueueInputPlugin plugin = new DeadLetterQueueInputPlugin(dir, false, null, new Timestamp(targetDateString));
         plugin.register();
     }
+
+    @Test
+    public void testClosingEmptyDlq() throws Exception {
+        Path since = temporaryFolder.newFile(".sincedb").toPath();
+        DeadLetterQueueInputPlugin plugin = new DeadLetterQueueInputPlugin(dir, true, since, null);
+
+        final AtomicInteger count = new AtomicInteger();
+        Thread pluginThread = new Thread(() -> {
+            try {
+                plugin.register();
+                plugin.run((e) -> {count.incrementAndGet();});
+            } catch (Exception e) {
+                // do nothing
+            }
+        });
+        pluginThread.start();
```

Comment:
I think the point here is to get the `pluginThread` after the `register`, to simulate a `run` that's not receiving data from DLQ because DLQ is empty.
So I think that using a `CountDownLatch` initialized to 1 and invoke the `countDown` after `register` would be more explicit in the meaning and avoid to leverage timing and depend OS scheduler in a not clear way.
For example:

```java
final CountDownLatch registered = new CountDownLatch(1);
...
try {
  plugin.register();
  plugin.run((e) -> {count.incrementAndGet();});
} 
....

pluginThread.start();
assertTrue("Can't get the input thread running after 1 second", registered.wait(1, TimeUnit.SECONDS));
```

Reply:
Instead of starting a thread and immediately after wait for it, I realized we don't need it at all. So in the latest push, the test just calls `plugin.register()` and then `plugin.close()`. This still demonstrates the bug / works with the fix, and there are no thread synchronization worries.